### PR TITLE
use undici agent

### PR DESCRIPTION
**Pull request by ronag**

### CODE REVIEW

**Comment by ronag** [comment]
@Ethan-Arrowood we need a `undici.Agent.close` and `undici.Agent.destroy` methods to make this work properly. Would you mind opening a PR over at undici for that?

**Comment by Ethan-Arrowood** [comment]
@ronag is there any way we could integrate an auto-close here?

**Comment by ronag** [comment]
> @ronag is there any way we could integrate an auto-close here?

How would that work? It does close now. But it takes `keepAliveTimeout` time.

**Comment by Ethan-Arrowood** [comment]
Hmm, ideally this fetch client should not have to have a `.close` method. I'd like to make the default behavior auto-close pools in some fashion.

I guess I can think of two scenarios:
- Long running process (such as an REST API) - should close unused pools after some time - if a new request comes through that uses a previous origin, that pool needs to be reopened (or just reinstantiated). Maybe we could consider a compounding timeout here where every time the pool timesout and is then asked for again, the following timeout time is increased.
- Short running process (such as a CLI command) - let the OS destroy it when the process ends execution (is this safe?). these should also have a timeout in case the "short" process runs for a long time, but we could set that timeout for a very long time and also integrate an auto-reset feature if it is reused after self-closing

**Comment by ronag** [comment]
The timeout is already there. I think this is mainly a problem for tests.

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
I think we should drop support for Node 10 and 12.

**Comment by Ethan-Arrowood** [comment]
I can't seem to push to your fork/branch @ronag but i have some changes to add here

**Comment by ronag** [comment]
I have checked "Allow edits by maintainers"

File: src/fetch.js [hunk 1]
```diff
@@ -5,6 +5,8 @@ const Request = require('./request')
 const Response = require('./response')
 const { STATUS_CODES } = require('http')
 
+const kAgent = Symbol('agent')
```
Comment (ronag) [inline src/fetch.js:8]: I would propose a symbols file which is not accessible for end users.

Comment (Ethan-Arrowood) [reply src/fetch.js:8]: okay and then add back the `.close` method to fetch in case a user wants to use that?

Comment (ronag) [reply src/fetch.js:8]: IMO expose a proper public api or make it entirely hidden.

**Comment by ronag** [comment]
undici now has agent, though probably needs an updated PR. Closing this.

**Pull request closed without merging**